packages/frontend/src/context/StoreContext.tsx

```diff
@@ -84,55 +85,33 @@ export const StoreContextProvider = (
     }
   }, [relayClient]);
 
-  // useEffect(() => {
-  //   (async () => {
-  //     const currencies = Array.from([...acceptedCurrencies.keys()]);
-  //     const _cur = currencies.filter((a) => !acceptedCurrencies.get(a));
-  //     _cur.map(async (address) => {
-  //       const { symbol } = await getTokenInformation(address);
-
-  //       setAcceptedCurrencies({
-  //         type: UPDATE_SYMBOL,
-  //         payload: { tokenAddr: address, symbol },
-  //       });
-  //     });
-  //   })();
-  // }, [acceptedCurrencies]);
-
-  // const verify = async (
-  //   _orderItems: Map<OrderId, OrderState>,
-  //   _pubKeys: `0x${string}`[],
-  // ) => {
-  //   console.log(_orderItems, _pubKeys);
-  // const addresses = _pubKeys.map((k) => {
-  //   return Address.fromPublicKey(hexToBytes(k)).toString();
-  // });
-  // const keysArr: `0x${string}`[] = _orderItems.size
-  //   ? Array.from([..._orderItems.keys()])
-  //   : [];
-  // for (const _orderId of keysArr) {
-  //   const _order = _orderItems.get(_orderId) as OrderState;
-  //   if (_order && _order.status !== Status.Failed) {
-  //     const sig = _order.signature as `0x${string}`;
-  //     const retrievedAdd = await relayClient!.recoverSignedAddress(
-  //       _orderId,
-  //       sig,
-  //     );
-  //     if (addresses.includes(retrievedAdd.toLowerCase())) {
-  //       console.log("inside inclue", _orderId);
-  //       setOrderId(_orderId);
-  //     }
-  //   }
-  // }
-  // };
-
   const getOrderId = async () => {
-    //FIXME: This part is still wip. We will have to go through the openOrders(array) and verify the signed address via keycard store. There should only be one open order per clerk.
-    // For now just set current order id as the first open order
+    //Getting all open orders.
     const openOrders = await stateManager?.orders.getStatus(Status.Pending);
-    let order_id: OrderId;
-    if (openOrders && openOrders.length) {
-      order_id = openOrders[0] as OrderId;
+    let order_id: OrderId | null = null;
+    if (openOrders && openOrders.length && walletAddress) {
+      //Getting all public keys by current wallet address.
+      const publicKeys = await stateManager?.keycards.get(
+        walletAddress.toLowerCase() as `0x${string}`,
+      );
+      //Getting address from public key.
+      const addresses = publicKeys.map((a: OrderId) => {
+        return Address.fromPublicKey(hexToBytes(a)).toString();
+      });
+      for (const o of openOrders) {
+        //Go through the open orders and see if the signer address is one of the public keys created with the wallet address.
```

Comment:
we shouldn't do this again and again when loading an order.

ideally we just check the signature once when we receive an event from the relay / before persisting it in the state manager. Then we would know that all events in the state manager where valid at the time of storing them. 

(clerks that were removed from a store are a separate problem we are tackling later.)

Reply:
I added line 90  `if (orderId) return orderId;` so this fixes the issue of having to calculate the orderId associated with the wallet address every time. 

In regards to this suggestion: _ideally we just check the signature once when we receive an event from the relay / before persisting it in the state manager._
I'm not sure where we would check the signature before updating the db. I think this would require us passing the wallet address to the stateManager, and then we still have the issue of somehow being able to access the keycard Store inside orderManager in order to get all the public keys associated with the WA in order to filter the correct orderId.

Reply:
> I'm not sure where we would check the signature before updating the db.

we should do it here
https://github.com/masslbs/Tennessine/blob/6c680f8e0b6c30c0f182cec29136491d284a12e8/packages/stateManager/index.ts#L641
before we call _processEvent on the objectStores